internal/controller/numaflowcontrollerrollout_controller_test.go

```diff
@@ -36,48 +42,102 @@ var _ = Describe("NumaflowControllerRollout Controller", func() {
+				restConfig: cfg,
+				rawConfig:  cfg,
+				stateCache: stateCache,
+				kubectl:    kubernetes.NewKubectl(),
+			}
 		})
 
 		AfterEach(func() {
-			// TODO(user): Cleanup logic after each test, like removing the resource instance.
 			resource := &apiv1.NumaflowControllerRollout{}
 			err := k8sClient.Get(ctx, typeNamespacedName, resource)
 			Expect(err).NotTo(HaveOccurred())
 
 			By("Cleanup the specific resource instance NumaflowControllerRollout")
 			Expect(k8sClient.Delete(ctx, resource)).To(Succeed())
 		})
-		// It("should successfully reconcile the resource", func() {
-		// 	By("Reconciling the created resource")
-		// 	controllerReconciler := &NumaflowControllerRolloutReconciler{
-		// 		client:     k8sClient,
-		// 		scheme:     k8sClient.Scheme(),
-		// 		restConfig: cfg,
-		// 	}
-
-		// 	_, err := controllerReconciler.Reconcile(ctx, sigsReconcile.Request{
-		// 		NamespacedName: typeNamespacedName,
-		// 	})
-		// 	Expect(err).NotTo(HaveOccurred())
-		// 	// TODO(user): Add more specific assertions depending on your controller's reconciliation logic.
-		// 	// Example: If you expect a certain status condition after reconciliation, verify it here.
-		// })
+
+		It("should successfully reconcile the resource", func() {
+			By("Reconciling the created resource")
+			result, err := controllerReconciler.Reconcile(ctx, sigsReconcile.Request{NamespacedName: typeNamespacedName})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).NotTo(BeNil())
+			Expect(result.Requeue).To(BeFalse())
+
+			// Validate the resource status
+			syncedResource := &apiv1.NumaflowControllerRollout{}
+			Expect(k8sClient.Get(ctx, typeNamespacedName, syncedResource)).To(Succeed())
+			Expect(syncedResource.Status.Phase).To(BeEquivalentTo("Running"))
+		})
+
+		It("should successfully reconcile the resource", func() {
+			By("Reconciling the updated resource")
+			resource := &apiv1.NumaflowControllerRollout{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-resource-updated",
+					Namespace: "default",
+				},
+				Spec: apiv1.NumaflowControllerRolloutSpec{
+					Controller: apiv1.Controller{Version: "1.2.0"},
+				},
+			}
+			typeNamespacedNameForUpdate := types.NamespacedName{
+				Name:      "test-resource-updated",
+				Namespace: "default",
+			}
+
+			// Create the resource to be updated
+			Expect(k8sClient.Create(ctx, resource)).To(Succeed())
+			result, err := controllerReconciler.Reconcile(ctx, sigsReconcile.Request{NamespacedName: typeNamespacedNameForUpdate})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).NotTo(BeNil())
+
+			// Fetch the resource and update the spec
+			var createdResource = &apiv1.NumaflowControllerRollout{}
+			Expect(k8sClient.Get(ctx, typeNamespacedNameForUpdate, createdResource)).To(Succeed())
+			createdResource.Spec = apiv1.NumaflowControllerRolloutSpec{
+				Controller: apiv1.Controller{Version: "1.2.1"},
+			}
+
+			// Update the resource after the spec change
```

Comment:
We also need the delete case, right? Or you want to have a separate PR for it?

Reply:
I'll raise a separate PR for it. Created an issue to track it: https://github.com/numaproj/numaplane/issues/46